polkadot/node/core/pvf/execute-worker/src/lib.rs

```diff
@@ -185,8 +189,7 @@ pub fn worker_entrypoint(
 
 				cfg_if::cfg_if! {
 					if #[cfg(target_os = "linux")] {
-						let stack_size = 2 * 1024 * 1024; // 2MiB
-						let mut stack: Vec<u8> = vec![0u8; stack_size];
+						let mut stack: Vec<u8> = vec![0u8; EXECUTE_THREAD_STACK_SIZE];
```

Comment:
We need to get this value from the executor params, and the constant should be removed. Please consult my earlier comment - there's a converter function you need to use on the exec params. Hope it makes sense. :pray:

Reply:
One question, is to replace the `EXECUTE_THREAD_STACK_SIZE` with `deterministic_stack_limit` of the `params_to_wasmtime_semantics`, correct? 
The `deterministic_stack_limit` is an Optional value, what is going to be the default value in case of `None` return?
Should we still use the `EXECUTE_THREAD_STACK_SIZE`?

Reply:
Yep, `deterministic_stack_limit.native_stack_max`!

Since we are spawning some threads in the process, we should add an additional 2 MiB (the default) to the process's stack size, for *each* thread's stack size. (Don't forget the process's main thread!) This should be documented, here as well as at the point where we spawn the threads, to make sure the value stays in sync with the number of threads.

In pseudo-code, `process_stack_size = wasm_native_stack_size + 2 MiB * num_threads`. Put another way, `process_stack_size = execute_stack_size + 2 MiB * num_other_threads`.*

The execute thread should continue to be spawned with 2 MiB (its own stack size) + the wasm stack size, as before. (And again, this should not be const.)

*NB, I am not sure if the process stack size is simply the sum of stack sizes of all threads, but it should be a good enough approximation for these purposes. It should suffice to account for the wasm stack size properly, as set in the executor params, and add a bit extra for the threads.

> The deterministic_stack_limit is an Optional value, what is going to be the default value in case of None return?

Good question! It's actually impossible for it to be `None` (the value comes from `DEFAULT_CONFIG`), and this is already [proven](https://github.com/paritytech/polkadot-sdk/blob/master/docs/contributor/STYLE_GUIDE.md#style) in `params_to_wasmtime_semantics`. I would return the `stack_limit` from `params_to_wasmtime_semantics` separately, since we are able to do so without wrapping it in `Option`, and then the function signature itself provides the guarantee that it is present.

Reply:
Thanks for the great explanation! Changed the stack size for the clone using 3 threads for the clone and 1 for the execute worker thread.